We need conda environment or manual installation to run the qtlseq analysis pipeline
1. Log in to Uppmax
2. check the envirionment
 conda env list
  453  conda install -c bioconda qtlseq
  454  conda env list
  455  conda deactivate
  456  conda env remove -n <environment_name> --all
  457  conda env remove -n /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda --all
  458  conda env list
  459  conda deactivate
  460  conda env list
  461  conda env remove -n base --all
  462  conda env remove -n base
  463  conda deactivate
  464  conda env remove -n base
  465  conda activate
  466  conda deactivate
  467  conda env remove -n base
  468  conda activate mamba
  469  conda env list
  470  conda env remove -n base --all
  471  conda env remove -n base
  472  conda install -c bioconda qtlseq
  473  conda install -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" "conda>=22.9"
  474  ll
  475  conda env list
  476  conda install -c bioconda qtlseq
  477  conda env remove -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda"
  478  conda env remove -n base -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda"
  479  conda env remove -n base -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" --all
  480  conda uninstall
  481  conda env remove 
  482  conda env list
  483  conda install -c bioconda qtlseq
  484  conda install -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" "conda>=22.9"
  485  conda env list
  486  conda create -n qtlseq python=3 qtlseq

   3. Git clone
  487  git clone https://github.com/YuSugihara/QTL-seq.git
  488  cd QTL-seq
  489  uquota
  490  du -sh /home/abusiddi
  491  ll
  492  ll -lrt
  493  ll -lrth
  494  rm quay.io-qiime2-core-2023.7.img.pulling.1709048715270
  495  ll -lrth
  496  git clone https://github.com/YuSugihara/QTL-seq.git
  497  ll -lrth
  498  cd QTL-seq
  499  pip install -e .
  500  ml pip
  501  cd QTL-seq
  502  ll
  503  ml bioifo-tools
  504  ml bioinfo-tools
  505  ml pip
  506  ml 
  507  ml python3
  508  ml
  509  pip install -e .
  510  pip install --upgrade pip
  511  pip install -e .
  512  qtlseq -h
  513  ll
  514  qtlseq
  515  ml qtlseq
  516  ml bioinfo-tools
  517  module spider qtlseq
  518  conda env list
  519  cd QTL-seq
  520  ll
  521  less README.md
  522  conda install -c bioconda qtlseq
  523  conda install -c mamba qtlseq
  524  conda update
  525  conda update --prefix /home/abusiddi/miniconda3
  526  conda update --prefix /home/abusiddi/miniconda3 anaconda
  527  uquota
  528  conda update --prefix /home/abusiddi/miniconda3 anaconda
  529  conda env list
  530  conda update --prefix /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda anaconda
  531  conda update --prefix /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda
  532  conda update -n base conda
  533  conda remove --all -n SqueezeMeta
  534  conda env list
  535  conda remove --all -n mamba
  536  conda env list
  537  conda remove --all -n qiime2-2018.11
  538  conda env list
  539  conda remove --all -n qiime2-2018.11
  540  conda env list
  541  conda remove --all -n base
  542  ll
  543  conda env list
  544  conda install -c bioconda qtlseq
  545  conda install -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" "conda>=4.11"
  546  conda upgrade -n base -c defaults --override-channels conda
  547  conda env list
  548  conda install -c bioconda qtlseq
  549  conda install -p "/home/abusiddi/miniconda3/ "conda>=4.11"
  550  conda install -p "/home/abusiddi/miniconda3/" "conda>=4.11"
  551  conda install -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" "conda>=4.11"
  552  conda remove --name conda --all command
  553  conda env list
  554  /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda
  555  ll /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda
  556  ll /sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda/envs/
  557  conda env list
  558  conda install -c miniconda3 qtlseq
  559  conda install -p "/sw/apps/bioinfo/qiime2/2018.11.0/rackham/conda" "conda>=4.11"
  560  Dconda upgrade -n base conda
  561  conda upgrade -n base conda
  562  conda env list
  563  exit
  564  conda env list
  565  ll
  566  ll -lrt
  567  cd QTL-seq/
  568  ll
  569  conda env list
  570  pip install -e .
  571  qtlseq
  572  conda install -c bioconda snpeff
  573  conda install -c bioconda trimmomatic
  574  ml trimmomatic
  575  ml
  576  ml bioinfo-tools
  577  ml trimmomatic
  578  ml
  579  ml snpEff
  580  ml
  581  snpEff --help
  582  trimmomatic --help
  583  qtlseq -h
 qtlseq -h 
  655  less darsync_vinitha.out
  656  cat darsync_vinitha.slurm
  657  ll /proj/snic2021-23-442/delivery05638
  658  ll /proj/snic2021-23-442/delivery05638/INBOX/UK-3126/220218_A00181_0431_AHN33MDSX2/
  659  less -S /proj/snic2021-23-442/delivery05638/INBOX/UK-3126/220218_A00181_0431_AHN33MDSX2/SampleSheet.csv
  660  ll /proj/snic2021-23-442/
  661  ll snpseq
  662  ll /proj/snic2021-23-442/
  663  ll /proj/snic2021-23-442/snpseq/data/
  664  du -sh /proj/snic2021-23-442/snpseq/data/
  665  du -sh /proj/snic2021-23-442/delivery05638
  666  cat darsync_vinitha.slurm
  667  nano darsync_vinitha.slurm
  668  ll /proj/snic2021-23-442/
  669  nano darsync_vinitha_snp.slurm
  670  ll
  671  nano darsync_vinitha_snp.slurm
  672  ll /home/abusiddi/id_ed25519_pdc
  673  cat /home/abusiddi/id_ed25519_pdc
  674  sbatch darsync_vinitha_snp.slurm
  675  nano darsync_vinitha_snp.slurm
  676  ll -lrth
  677  jobinfo -u abusiddi
  701  /proj/uppstore2018171/abu/vinitha_qtl/
  702  ll /proj/uppstore2018171/abu/vinitha_qtl/
  703  du -sh /proj/uppstore2018171/abu/vinitha_qtl/snpseq/
  704  du -sh /proj/uppstore2018171/abu/vinitha_qtl/snpseq/data/
  705  tree /proj/uppstore2018171/abu/vinitha_qtl/snpseq/data/
  706  cat /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_r2.err 
  707  uquota
  711  nano darsync_vinitha_snp.slurm
  712  cat darsync_vinitha_snp.slurm
  713  nano trsnsfer_qtlseq.sh
  714  cat darsync_vinitha_snp.slurm
  715  uquota
  716  ll /proj/uppstore2018171/abu/vinitha_qtl/
  717  projinfo
  718  ll
  719  ll -lrth 
  720  less darsync_vinitha_snp.err
  721  jobinfo -u abusiddi
  722  ll .ssh/
  723  ll .ssh/known_hosts 
  724  ll .ssh/
  725  less .ssh/known_hosts 
  726  chmod 700 .ssh
  727  ll .ssh/
  728  ssh-keygen -t ed25519 -f ~/.ssh/id-ed25519-pdc
  729  ssh -i ~/.ssh/id-ed25519-pdc <username-at-pdc>@dardel.pdc.kth.se
  730  ssh -i ~/.ssh/id-ed25519-pdc absiddi@dardel.pdc.kth.se
  731  less .ssh/id-ed25519-pdc.pub
  732  ssh -i ~/.ssh/id-ed25519-pdc absiddi@dardel.pdc.kth.se
  733  nano darsync_vinitha_snp.slurm
  734  sbatch darsync_vinitha_snp.slurm
  735  jobinfo -u abusiddi
  736  ll -lrth 
  737  cat darsync_vinitha_snp.err
  738  ml
  739  nano darsync_vinitha_snp.slurm
  740  ml darsync
  741  ml
  742  sbatch darsync_vinitha_snp.slurm
  743  jobinfo -u abusiddi
  744  cat darsync_vinitha_snp.err
  745  ml darsync
  746  darsync check
  747  ll /proj/snic2021-23-442/snpseq/
  748  less /proj/snic2021-23-442/snpseq/darsync_snpseq.ownership.gz
  749  uquota
  750  ll /proj/uppstore2018171/abu/
  751  cd /proj/uppstore2018171/abu/
  752  mkdir vinitha_qtl
  753  cd vinitha_qtl/
  754  ll
  755  pwd
  756  nano qtlseq_transfer.sh
  757  ll
  758  ls qtlseq_transfer.sh 
  759  nano qtlseq_transfer.sh
  760  sbatch qtlseq_transfer.sh
  761  jobinfo -u abusiddi
  762  ll
  763  ll -h snpseq/
  764  less qtlseq.err
  765  nano qtlseq_transfer.sh
  766  less qtlseq.out
  767  nano qtlseq_transfer.sh
  768  sbatch qtlseq_transfer.sh
  769  jobinfo -u abusiddi
  770  scancel 47046886
  771  nano qtlseq_transfer.sh
  772  sbatch qtlseq_transfer.sh
  773  jobinfo -u abusiddi
  774  uquota
  775  cat darsync_vinitha_snp.slurm 
  776  ll /proj/snic2021-23-442/snpseq/ 
  777  du -sh /proj/snic2021-23-442/snpseq/

  cd /proj/uppstore2018171/abu/vinitha_qtl/
  nano qtlseq_transfer.sh
  
sbatch qtlseq_transfer.sh
ll -lrth

cd /proj/uppstore2018171/abu/vinitha_qtl/
less qtlseq_r2.out
du -sh snpseq/

qtlseq

 ml bioinfo-tools snpEff
 
 nano qtlseq_transfer.sh

 4. test run

 cd /proj/uppstore2018171/abu/vinitha_qtl/
 
mkdir test_run
test_run/

git clone https://github.com/YuSugihara/QTL-seq.git

ll QTL-seq/
ll QTL-seq/test/
QTL-seq/test/qtlseq.sh 
nano  QTL-seq/test/qtlseq.sh 
cat ../qtlseq_transfer.sh 
nano  QTL-seq/test/qtlseq.sh 
nano  QTL-seq/test/qtlseq_test1.sh 
ll QTL-seq/test/
ll-h  QTL-seq/test/
ll -h QTL-seq/test/

qtlseq
nano  QTL-seq/test/qtlseq_test1.sh 
cd QTL-seq/test/

sbatch qtlseq_test1.sh

jobinfo -u abusiddi

ll ../
tree test_1/
cat qtlseq_test1.sh 
less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_r2.err
less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_r2.out 
nano  QTL-seq/test/qtlseq_test1.sh 
nano qtlseq_test1.sh 

sbatch qtlseq_test1.sh
jobinfo -u abusiddi

cat qtlseq_r2.err

# error, test run finished unsuccessfully, now run again

rm -r test_1

sbatch qtlseq_test1.sh
cat qtlseq_r1.out

tree test_1/
jobinfo -u abusiddi

# error, run again

rm -r test_1
nano qtlseq_test1.sh 
sbatch qtlseq_test1.sh

jobinfo -u abusiddi
du -sh test_1/
du -sh ./
ll -h

tree ../../../snpseq/

less ../../../snpseq/data/files/WJ-3814/20240220_LH00179_0046_B22HWMNLT3/README.md 
  878  less ../../../snpseq/data/files/WJ-3814/20240220_LH00179_0046_B22HWMNLT3/SampleSheet.csv 
  879  projinfo
  880  ml bioinfo-tools snpEff trimmomatic bwa samtools bcftools

  882  qtlseq
  883  ll
  884  cd qtlseq_analysis/
  885  ll
  886  sbatch qtlseq_run1.sh
  887  ll
  888  jobinfo -u abusiddi
  889  ll
  890  less qtlseq_run2.err
  891  less qtlseq_run2.out
  892  less qtlseq_run1.out
# run successfully, now real run with SB genome, download the genome files  

893  gzip -dk Phytozome/PhytozomeV13/Bvulgarisssp_vulgaris/EL10.2_2/assembly/Bvulgarisssp_vulgaris_782_EL10.2.fa.gz 
  894  gzip -d -k Phytozome/PhytozomeV13/Bvulgarisssp_vulgaris/EL10.2_2/assembly/Bvulgarisssp_vulgaris_782_EL10.2.fa.gz
  895  gzip -dk Phytozome/PhytozomeV13/Bvulgarisssp_vulgaris/EL10.2_2/assembly/Bvulgarisssp_vulgaris_782_EL10.2.fa.gz .
  896  cp Phytozome/PhytozomeV13/Bvulgarisssp_vulgaris/EL10.2_2/assembly/Bvulgarisssp_vulgaris_782_EL10.2.fa.gz .
  897  ll
  898  ll -h
  899  gunzip Bvulgarisssp_vulgaris_782_EL10.2.fa.gz
  900  ll
  901  ll -h

  # run sbatch
  902  pwd
  903  sbatch qtlseq_run2.sh
  904  jobinfo -u abusiddi
  905  ll qtlseq_run_2
  906  ll qtlseq_analysis_run_2
  907  jobinfo -u abusiddi
  908  ll qtlseq_analysis_run_2
  909  tree qtlseq_analysis_run_2
  910  ll
  911  less qtlseq_run2.err
  912  jobinfo -u abusiddi
  913  projinfo
  914  sbatch qtlseq_run3.sh
  915  ml
  916  ml snpEff
  917  jobinfo -u abusiddi
  918  ll
  919  cd ../..
  920  cd ..
  921  ll
  922  mkdir qtlseq_analysis
  923  cd qtlseq_analysis/
  924  ll
  925  pwd
  926  curl --cookie jgi_session=/api/sessions/07018f619c8c3de104b2e2c974653a90 --output download.20240520.130135.zip -d "{\"ids\":{\"Phytozome-782\":[\"6334df862de7c323533ed952\",\"6334df862de7c323533ed95c\",\"6334df862de7c323533ed956\",\"6334df872de7c323533ed970\",\"6334df872de7c323533ed96a\",\"6334df872de7c323533ed962\",\"6334df872de7c323533ed96c\",\"6334df862de7c323533ed95e\",\"6334df862de7c323533ed954\",\"6334df862de7c323533ed95a\",\"6334df872de7c323533ed966\",\"6334df872de7c323533ed968\",\"6334df862de7c323533ed958\",\"6334df872de7c323533ed964\",\"6334df872de7c323533ed960\",\"6334df872de7c323533ed972\",\"6334df872de7c323533ed96e\",\"6334df862de7c323533ed950\"]}}" -H "Content-Type: application/json" https://files-download.jgi.doe.gov/filedownload/
  927  ll
  928  unzip download.20240520.130135.zip
  929  ll
  930  tree Phytozome/
  931  cat ../test_run/QTL-seq/test/qtlseq_test1.sh
  932  nano qtlseq_run1.sh
  933  jobinfo -u abusiddi
  934  ll -h qtlseq_analysis/qtlseq_analysis_run_3/00_fastq/
  935  uquota
  936  ll -h qtlseq_analysis/qtlseq_analysis_run_3/00_fastq/
  937  scancel 47108229
  938  ll
  939  jobinfo -u abusiddi
  940  ll
  941  cd qtlseq_analysis/
  942  ll
  943  sbatch qtlseq_run4.sh 
  944  jobinfo -u abusiddi
  945  exit
  946  jobinfo -u abusiddi-
  947  history
  948  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  949  ll /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/00_fastq/
  950
      ll /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_3/
  956  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_3/
  957  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  958  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  959  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  960  less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_run4.out 
  961  less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_run4.err 
  962  jobinfo -u abusiddi
  963  less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_run4.err 
  964  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  965  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  966  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/log/trimmomatic.log 
  967  cat /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/log/trimmomatic.log 
  968  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  969  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  970  less /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_run4.out 
  971  cat /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/log/alignment.log
  972  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/log/
  973  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  974  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/00_fastq/
  975  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  976  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/log
  977  jobinfo -u abusiddi
  978  tree /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/
  979  ll -h /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/00_fastq/
  980  ll /proj/uppstore2018171/abu/vinitha_qtl/qtlseq_analysis/qtlseq_analysis_run_4/00_fastq/
  981  ll
  982  pwd
  983  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly
  984  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/output_anno
  985  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/output_annotation_nano_assembly/Taxonomy/
  986  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/output_annotation_nano_assembly/Taxonomy/kraken2/
  987  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/output_annotation_nano_assembly/Assembly/
  988  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/output_annotation_nano_assembly/Assembly/metaflye/
  989  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/
  990  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/
  991  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/out_nano_metaflye/
  992  ll /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/out_nano_metaflye/kraken_assembly_output/
  993  less /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/metatdenovo/results_metatdenovo_mc1000_full_r2/eggnog/user_assembly.prodigal.emapper.annotations.gz
  994  less -S /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/metatdenovo/results_metatdenovo_mc1000_full_r2/eggnog/user_assembly.prodigal.emapper.annotations.gz
  995  less l
  996  less -S /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/metatdenovo/results_metatdenovo_mc1000_full_r2/kofamscan/user_assembly.prodigal.kofamscan_output.tsv.gz
  997  less -S /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/metatdenovo/results_metatdenovo_mc1000_full_r2/prodigal/user_assembly.prodigal_all.txt.gz
  998  less -S /proj/snic2022-23-615/nobackup/private/mag/nxf/sqzmeta/kong_metagenom/annotation_nano_assembly/metatdenovo/results_metatdenovo_mc1000_full_r2/prodigal/user_assembly.prodigal.gff.gz
